[new-product] opnsense (#4385)
captn3m0/endoflife.date: products/opnsense.md

---
title: OPNsense
category: server-app
permalink: /opnsense
changelogTemplate: "https://docs.opnsense.org/releases/CE___RELEASE_CYCLE__.html"
releaseDateColumn: true
iconSlug: opnsense
tags: php-runtime
releasePolicyLink: https://opnsense.org/about/road-map/

> releaseLabel: "__RELEASE_CYCLE__ '__CODENAME__'"

Reviewer: ```suggestion
title: OPNsense
category: server-app
tags: php-runtime
iconSlug: opnsense
permalink: /opnsense
releasePolicyLink: https://opnsense.org/about/road-map/
changelogTemplate: "https://docs.opnsense.org/releases/CE___RELEASE_CYCLE__.html"
releaseLabel: "__RELEASE_CYCLE__ '__CODENAME__'"
releaseDateColumn: true
```

Applied style guide: https://github.com/endoflife-date/endoflife.date/wiki/Style-Guide.

Author: applied, ty, I will take a look on my other PR

auto:
-   git: https://github.com/opnsense/core
    # https://regex101.com/r/BV8amT/2

>     regex: ^(?<major>\d+\.\d+)\.(?<minor>\d{1,3})\.?(?<patch>\d+)?$

Reviewer: ```suggestion
```

Except if there are some minor with more than 3 digits that must be ignored, [the default regex](https://github.com/endoflife-date/release-data/blob/62e1c4c0a969b8d6caba1b4d91a1d22f642a3483/src/common/endoflife.py#L15) should be enough.

Author: applied, ty

# eol(x) = releaseDate(x+1)
releases:

# -   releaseCycle: "24.1"
#    releaseDate: 2024-01-XX
#    eol: false
#    latest: "24.1"
#    latestReleaseDate: 2024-01-XX
#    codename: "YYYYYYYY"

> 

Reviewer: ```suggestion
```

Previous releases can be use as a template too.

Author: applied

-   releaseCycle: "23.7"
    releaseDate: 2023-07-31
    eol: false
    latest: "23.7.10"
    latestReleaseDate: 2023-12-12

>     codename: "Restless Roadrunner"

Reviewer: ```suggestion
-   releaseCycle: "23.7"
    codename: "Restless Roadrunner"
    releaseDate: 2023-07-31
    eol: false
    latest: "23.7.10"
    latestReleaseDate: 2023-12-12
```

Can you move the `codename` just after the `releaseCycle` for all releases ? See https://github.com/endoflife-date/endoflife.date/wiki/Style-Guide.

Author: applied

    latestReleaseDate: 2018-07-24
    codename: "Groovy Gecko"

-   releaseCycle: "17.7"
    releaseDate: 2017-07-31
    eol: 2018-01-29
    latest: "17.7.12"
    latestReleaseDate: 2018-01-18
    codename: "Free Fox"

-   releaseCycle: "17.1"
    releaseDate: 2017-01-31
    eol: 2017-07-31
    latest: "17.1.11"
    latestReleaseDate: 2017-07-25
    codename: "Eclectic Eagle"

-   releaseCycle: "16.7"
    releaseDate: 2016-07-28
    eol: 2017-01-31
    latest: "16.7.14"
    latestReleaseDate: 2017-01-25
    codename: "Dancing Dolphin"

-   releaseCycle: "16.1"
    releaseDate: 2016-02-02
    eol: 2016-07-28
    latest: "16.1.20"
    latestReleaseDate: 2016-07-22
    codename: "Crafty Coyote"

-   releaseCycle: "15.7"
    releaseDate: 2015-07-02
    eol: 2016-02-02
    latest: "15.7.25"
    latestReleaseDate: 2016-01-18
    codename: "Brave Badger"

-   releaseCycle: "15.1"
    releaseDate: 2015-01-02
    eol: 2015-07-02
    latest: "15.1.12"
    latestReleaseDate: 2015-06-07
    codename: "Ascending Albatross"
    
---

> [OPNsense](https://opnsense.org/) is an open-source firewall and routing platform with a user-friendly interface and robust features for network security.

> The only officially supported and actively maintained version of OPNsense is the latest one.  
> 
> ## [Release Schedule](https://docs.opnsense.org/manual/updates.html)
> 
> | Version Type         | Description                                                                                           | Cadence        |
> |:---------------------|:------------------------------------------------------------------------------------------------------|:---------------|
> | Major                | For significant changes, will include breaking or backwards-incompatible changes.                     | Every 6 months |
> | Minor                | Will include new features, bug fixes and security patches.                                            | Every 15 days  |

Reviewer: ```suggestion
OPNsense follows a [Calendar Versioning scheme](https://docs.opnsense.org/manual/updates.html#updates) with two major releases per year : one in January and the other in July. [Only the latest release is supported](https://docs.opnsense.org/releases/CE_23.7.html#january-04-2024) with one update every two weeks containing new non-breaking and backward-compatible features, bug fixes and security updates.
```

Much more succinct IMO, WDYT ?

Also added a link confirming that only the latest release is supported:

> The final test phase for 24.1 is starting just as 23.7 strechtes towards its inevitable end of life.

Author: LGTM
